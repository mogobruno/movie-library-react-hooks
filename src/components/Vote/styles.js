export default theme => ({
  content: {
    position: 'relative',
    minWidth: '44px',
    maxWidth: '44px',
    height: '44px',
    borderRadius: '100%',
    textAlign: 'center',
    paddingTop: '12px',
    backgroundColor: '#081c22',
    color: '#ffffff'
  },
  canvas: {
    position: 'absolute',
    top: '-2px',
    left: '-2px'
  }
})
