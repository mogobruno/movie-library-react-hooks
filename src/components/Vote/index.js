import PropTypes from 'prop-types'
import React, { useEffect } from 'react'
import { withStyles } from '@material-ui/core/styles'

// Services
import radiansFromPercentage from 'services/radiansFromPercentage'
import circleColor from 'services/circleColor'

// Styles
import styles from './styles'

const Vote = ({ classes, value }) => {
  let canvas = React.createRef()

  useEffect(() => {
    const ctx = canvas.current.getContext("2d")
    if (ctx) {
      // Draws full circle
      ctx.lineCap = "round"
      ctx.beginPath()
      ctx.strokeStyle = "#223135"
      ctx.lineWidth = 3
      ctx.arc(24, 24, 18, 0, 2 * Math.PI)
      ctx.stroke()

      // Draws percentage circle
      ctx.beginPath()
      ctx.lineWidth = 3
      ctx.strokeStyle = circleColor(value)
      ctx.arc(24, 24, 18, 0, radiansFromPercentage(value * 10) * Math.PI)
      ctx.stroke()
    }
  })

  return (
    <div className={classes.content}>
      <span className={classes.content}>{value}</span>
      <canvas className={classes.canvas} ref={canvas}></canvas>
    </div>
  )
}

Vote.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(Vote)
