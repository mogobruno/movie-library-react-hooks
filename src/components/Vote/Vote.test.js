import 'jest-canvas-mock'

import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Vote from './index'

describe('<Vote />', () => {

  let component

  beforeEach(() => {
    component = (
      <Vote value={7.4} />
    )
  })

  it('allows us to set props', () => {
    const wrapper = mount(component)
    expect(wrapper.props().value).toBe(7.4)
    wrapper.setProps({ value: 5.3 })
    expect(wrapper.props().value).toBe(5.3)
  })

  test('toMatchSnapshot', () => {
    const treeEmptyList = renderer
      .create(component)
      .toJSON()
    expect(treeEmptyList).toMatchSnapshot()
  })

})
