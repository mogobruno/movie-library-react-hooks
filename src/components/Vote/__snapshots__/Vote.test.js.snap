// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vote /> toMatchSnapshot 1`] = `
<div
  className="Vote-content-1"
>
  <span
    className="Vote-content-1"
  >
    7.4
  </span>
  <canvas
    className="Vote-canvas-2"
  />
</div>
`;
